src/process.cr

```diff
@@ -55,10 +55,11 @@ class Process
   #
   # The handler is executed on a fresh fiber every time an interrupt occurs.
   #
-  # * On Unix-like systems, this traps `SIGINT`.
-  # * On Windows, this captures <kbd>Ctrl</kbd> + <kbd>C</kbd> and
-  #   <kbd>Ctrl</kbd> + <kbd>Break</kbd> signals sent to a console application.
-  def self.on_interrupt(&handler : ->) : Nil
+  # * On Unix-like systems, this traps `SIGINT`, `SIGHUP`, `SIGTERM`.
+  # * On Windows, this captures <kbd>Ctrl</kbd> + <kbd>C</kbd>,
+  #   <kbd>Ctrl</kbd> + <kbd>Break</kbd>, terminal close, windows logoff
+  #   and shutdown signals sent to a console application.
+  def self.on_interrupt(&handler : ::Process::ExitReason ->) : Nil
```

Comment:
Unfortunately, this is a breaking change. Existing calls might be incompatible with the new signature.

This is not a problem when the proc argument is a directly captured block (`Process.interrupt do ... end`). Implicit autocasting creates the appropriate proc type.

But calls passing a proc value would fail due to mismatching types:
```cr
my_handler = -> { }
Process.on_interrupt(&my_handler) # Error: wrong number of block argument's parameters (given 0, expected 1)
```

Reply:
Could a macro be used to detect if the block has parameters or not?
Happy to attempt that workaround

Reply:
okay I should have solved this issue